packages/generator/ni_measurement_plugin_sdk_generator/client/templates/measurement_plugin_client.py.mako

```diff
@@ -0,0 +1,108 @@
+<%page args="class_name, measure_docstring, configuration_metadata, output_metadata, service_class, configuration_parameters_with_type_and_default_values, measure_api_parameters, output_parameters_with_type, import_modules"/>\
+\
+"""Python Measurement Plug-In Client."""
+
+% for module, import_type in import_modules.items():
+% if import_type == "Built_In":
+${module}
+% endif
+% endfor
+from typing import List, NamedTuple
+
+import grpc
+from google.protobuf import any_pb2
+from ni_measurement_plugin_sdk_service._internal.parameter.metadata import ParameterMetadata
+from ni_measurement_plugin_sdk_service._internal.stubs.ni.measurementlink.measurement.v2 import (
+    measurement_service_pb2 as v2_measurement_service_pb2,
+    measurement_service_pb2_grpc as v2_measurement_service_pb2_grpc,
+)
+% for module, import_type in import_modules.items():
+% if import_type == "Custom":
+${module}
+% endif
+% endfor
+from ni_measurement_plugin_sdk_service.discovery import DiscoveryClient
+from ni_measurement_plugin_sdk_service.grpc.channelpool import GrpcChannelPool
+
+from ni_measurement_plugin_sdk_generator.parameter import (
+    create_file_descriptor,
+    deserialize_parameters,
+    serialize_parameters,
+)
```

Comment:
I don't think running the output of the generator should require the generator to be installed. Run time support code should go into either the service package or a new package.

Reply:
Sure. 
The methods are already present in the service package under `_internals/parameter`. Based on the [comment](https://github.com/ni/measurement-plugin-python/pull/841#discussion_r1718972263), I have created a wrapper for those methods and exposed it as public.

Instead of moving these wrappers to the service package, should we consider moving `decode, encode, and serialization_descriptors` out of _internal? I would like to know your thoughts on this.

Reply:
We haven't put much thought into insulating those classes from interface changes. I think you should expose only the functionality you need, and expose it in a way that makes it clear that it is only for supporting client codegen.

Reply:
I have created a sub-package for methods used in `_internal` - `decode, encode, serialization_descriptors, ParameterMetadata`, under `service`.